### PR TITLE
feat: Transcode non-UTF-8 feeds to UTF-8 before XMLParser

### PR DESCRIPTION
**Pull request by nicholas-lonsinger**
## Summary
- Feed payloads in UTF-16, UTF-32, Big5, EUC-KR, GB2312, ISO-8859-*, Windows-1252, and other IANA-registered charsets now parse correctly. `XMLParser` only reliably handles UTF-8/ASCII, so previously these feeds either errored outright or silently surfaced garbled text (e.g., the Korean `년` character in a real EUC-KR fixture came through as `��`).
- A new `EncodingSniffer` pre-processing pass runs on every `parse(_:)` call with a zero-allocation fast path for UTF-8 (the 95% case). Detection precedence matches the XML spec: BOM wins over declaration, and UTF-32 is checked before UTF-16 to avoid `FF FE 00 00` being misclassified as UTF-16 LE with trailing nulls.

## Why this scope and not more
The initiative started as a broader "harden the RSS/Atom parser against messy feeds" project, seeded by feedparser's ~2,100-fixture test corpus. I ran the existing parser against the full corpus as a baseline **before** writing any fixes. Results:

- **Date parsing:** 43 of 44 unique date strings in the corpus parsed (97.7%). The one failure was a deliberately-illformed Greek ISO-8859-7 fixture in `illformed/` that feedparser itself only tests for "don't crash" behavior. My original 8-point date hardening plan (full month names, extra commas, asctime, W3DTF fragments, US single-letter zones) was falsified — those formats appear in feedparser's Python self-tests, not in real feed payloads. Date hardening was dropped as solution-in-search-of-a-problem.
- **Encoding detection:** ~40 `encoding/` fixtures failed due to UTF-16/UTF-32/BOM handling. This is the one real gap and the target of this PR.
- **Custom namespace prefixes on `<rss>` root:** ~4 fixtures, documented out-of-scope in the existing RATIONALE comment, deliberately skipped.
- **Empty format-detection fixtures / CDF / illformed:** correct rejections, not real problems.

## Changes
- **`RSSApp/Services/RSSParsingService.swift`**: Add internal `EncodingSniffer` enum (`transcodeToUTF8IfNeeded`, `detectBOM`, `detectWideEncodingWithoutBOM`, `scanEncodingDeclaration`, `encodingFromIANAName`, `transcode`, `stripProlog`, `stripXMLDeclaration`). `parse(_:)` now calls `EncodingSniffer.transcodeToUTF8IfNeeded(data)` before handing bytes to `XMLParser`. When transcoding happens, the original `<?xml ... ?>` prolog is stripped from the UTF-8 output so `XMLParser` doesn't see a stale `encoding="big5"` attribute that contradicts the actual bytes. Unknown encoding names fall back to a byte-level prolog strip + UTF-8 default (best-effort recovery for feeds that typo their encoding declaration). IANA charset names are resolved via `CFStringConvertIANACharSetNameToEncoding`, which covers every charset the system understands.
- **`RSSAppTests/Services/RSSParsingEncodingTests.swift`** (new, 19 tests): Inline-byte fixtures (no filesystem dependency) for every branch of the sniffer — plain UTF-8, UTF-8 with BOM, UTF-16 LE/BE with and without BOM, ISO-8859-1 and Windows-1252 via declaration, single- vs double-quoted encoding attributes, unknown charset name (graceful degradation via `stripProlog`), empty data, plus focused unit tests for `detectBOM`, `detectWideEncodingWithoutBOM`, `scanEncodingDeclaration`, `encodingFromIANAName`, `stripXMLDeclaration`, and the leading-U+FEFF edge case.
- **Drive-by `nonisolated(unsafe)` cleanup**: Remove four annotations the compiler now flags as unnecessary. `HoistedDateFormatters.zoned` and `.zoneless` in `RSSParsingService.swift` (the `[(String, DateFormatter)]` tuple array is Sendable on its own; the pre-existing RATIONALE comment is amended to drop the "nonisolated(unsafe) is safe because..." opener while preserving the performance/concurrency reasoning). `MockHTMLURLSessionProvider.storeQueue` and `MockSSEURLProtocol.storeQueue` in the test mocks (DispatchQueue is now Sendable). The `requestStore` vars in those mocks and both ISO8601 formatter annotations stay untouched — they are genuinely mutable or their container type isn't Sendable.
- **`ARCHITECTURE.md`**: Update the `RSSParsingService.swift` directory entry and add a `RSSParsingEncodingTests.swift` test-tree entry. Add a Design Decisions row explaining the sniffer's detection precedence and the BOM-over-declaration rationale.

## Evidence
**Corpus wins (per-file audit against the feedparser encoding/ fixtures):**
- `euc-kr-item.xml` → description `'년'` (was `'��'`)
- `euc-kr-attribute.xml` → description `'년'`
- `http_encoding_attribute_crash.xml` → smart quote `'‘'` decoded from windows-1252
- `http_text_xml_charset_overrides_encoding_2.xml` → `'£'` decoded from ISO-8859-1

Fixtures still reporting `noChannelFound` in `encoding/` after the fix are almost all empty `<rss></rss>` format-detection probes (same intentional-empty category as `wellformed/rss/rss_version_20.xml`), not real regressions.

## Test plan
- [x] Built successfully for iOS 26
- [x] Full test suite passes: 905 tests green, 0 failing (+19 from the new encoding suite)
- [x] All 139 pre-existing `RSSParsingServiceTests` still pass unchanged
- [x] Compiler warnings for `nonisolated(unsafe)` on the four flagged sites are gone
- [x] Rebased cleanly onto current main (post-#270); full suite green on rebased state

## Notes
- The feedparser corpus smoke tests I used to measure the baseline are **not** checked in. They depended on `/tmp/feedparser-corpus` being staged separately and would be dead code for other contributors. Regression coverage for every encoding variant the corpus exposed lives in the new hermetic test file.
- Retaining this note as a process lesson: the corpus baseline falsified my original 8-point date hardening plan before I wrote any code. "Which gaps are real" is worth measuring before being fixed.

### CODE REVIEW

File: RSSAppTests/Services/RSSParsingEncodingTests.swift [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+        <rss version="2.0"><channel><title>Pound</title><item><title>Cost</title><description>A £ test</description></item></channel></rss>
+        """
+        guard let bodyData = body.data(using: .isoLatin1) else {
+            Issue.record("Could not encode body as ISO-8859-1")
+            return
+        }
+        let feed = try service.parse(head + bodyData)
+        #expect(feed.articles.first?.articleDescription == "A £ test")
+    }
```
[the remark this answers is not in the document]
Comment (nicholas-lonsinger) [reply RSSAppTests/Services/RSSParsingEncodingTests.swift]: Dismissing — this is a false positive. The prolog `head` is pure ASCII, which is byte-identical in UTF-8 and ISO-8859-1 (both map 0x00–0x7F to the same scalars), so concatenating ASCII-bytes + ISO-8859-1-body produces a byte sequence indistinguishable from a full ISO-8859-1 document. The test exercises the full pipeline (declaration scan → IANA lookup → decode → strip → re-encode → parse) correctly. I did add a dedicated end-to-end Big5 test in e1785d0 that uses `xml.data(using: big5)` for the whole document — that covers the CJK case where the all-in-one encoding construction actually matters.
